firedrake/interpolation.py

```diff
@@ -228,18 +238,18 @@ def _interpolator(V, tensor, expr, subset, arguments, access):
     if not isinstance(expr, firedrake.Expression):
         if expr.ufl_domain() and expr.ufl_domain() != V.mesh():
             raise NotImplementedError("Interpolation onto another mesh not supported.")
-        ast, oriented, needs_cell_sizes, coefficients, _ = compile_expression_dual_evaluation(expr, to_element, coords,
-                                                                                              domain=V.mesh(), coffee=False)
-        kernel = op2.Kernel(ast, ast.name, requires_zeroed_output_arguments=True)
+        ast, oriented, needs_cell_sizes, coefficients, _ = compile_expression_dual_evaluation(expr, element, coords, coffee=False)
+        kernel = op2.Kernel(ast, ast.name)
     elif hasattr(expr, "eval"):
         to_pts = []
-        for dual in to_element.fiat_equivalent.dual_basis():
+        # For FIAT elements
+        for dual in element.dual_basis():
```

Comment:
I think this line, with `element` renamed back to to `to_element` is the only change needed in `firedrake/interpolation.py` thanks to the changes introduced in #1818

Reply:
👍 for throwing away unnecessary changes and/or superfluous if-then-else jungle.